cql3/statements/alter_keyspace_statement.cc

```diff
@@ -198,8 +198,13 @@ cql3::statements::alter_keyspace_statement::prepare_schema_mutations(query_proce
         auto ks_md_update = _attrs->as_ks_metadata_update(ks_md, tm, feat);
         std::vector<mutation> muts;
         std::vector<sstring> warnings;
-        bool include_tablet_options = _attrs->get_map(_attrs->KW_TABLETS).has_value();
+        // _attrs->get_map(_attrs->KW_TABLETS) might not exist when `AND tablets = {...}` was not used in the
+        // ALTER KEYSPACE statement, but we want to include this information from the metadata if tablets are
+        // enabled.
+        bool include_tablet_options = ks_md->initial_tablets().has_value();
```

Comment:
How about ditching the `include_tablet_options` variable?
Can't `get_old_options_flattened` and `get_current_options_flattened` check that on their own?
We pass ks to `get_old_options_flattened`, so it has access to `ks.metadata()`
and we pass `_attr` to `get_current_options_flattened`, so it see if the KW_TABLETS map is provided or not.

Reply:
Even better.